File: NewsDaily/app/src/main/java/com/andruid/magic/newsdaily/articles/ArticlesDataSourceFactory.java
package com.andruid.magic.newsdaily.articles;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;
import androidx.paging.PageKeyedDataSource;

import com.andruid.magic.newsloader.model.News;

public class ArticlesDataSourceFactory extends DataSource.Factory<Integer, News> {
    private MutableLiveData<PageKeyedDataSource<Integer, News>> liveDataSource =
            new MutableLiveData<>();
    private String language, query;


    ArticlesDataSourceFactory(String language, String query){
        this.language = language;
        this.query = query;
    }

    @NonNull
    @Override
    public DataSource<Integer, News> create() {
        ArticlesDataSource dataSource = new ArticlesDataSource(language, query);
        liveDataSource.postValue(dataSource);
        return dataSource;
    }
}

File: NewsDaily/app/src/main/java/com/andruid/magic/newsdaily/articles/ArticlesDataSourceFactory.kt
package com.andruid.magic.newsdaily.articles

import androidx.paging.DataSource
import com.andruid.magic.newsloader.model.News

class ArticlesDataSourceFactory(val language : String, val query : String) : DataSource.Factory<Int, News>() {

    override fun create(): DataSource<Int, News> {
        return ArticlesDataSource(language, query)
    }
}